light-system-programs/programs/verifier_program_storage/src/lib.rs

```diff
@@ -90,20 +90,24 @@ pub mod verifier_program_storage {
             InstructionDataShieldedTransferSecond::try_deserialize_unchecked(
                 &mut [vec![0u8; 8], inputs, vec![0u8; 16]].concat().as_slice(),
             )?;
-        let message = &ctx.accounts.verifier_state.msg;
-        let message_hash = hash(message).to_bytes();
+        let message = Message::new(&ctx.accounts.verifier_state.msg);
+        let proof = Proof {
+            a: inputs.proof_a,
+            b: inputs.proof_b,
+            c: inputs.proof_c,
+        };
+        let public_amount = Amount {
+            sol: inputs.public_amount_sol,
+            spl: [0u8; 32],
+        };
 
         process_shielded_transfer_2_in_2_out(
             &ctx,
-            Some(&message_hash),
-            Some(message),
-            &inputs.proof_a,
-            &inputs.proof_b,
-            &inputs.proof_c,
-            &[0u8; 32], // Verifier storage does not support SPL tokens.
```

Comment:
Probably makes sense to keep the comment, I was a bit confused at first.

Reply:
Added it back